src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -220,6 +220,12 @@ func (r *Cluster) validateAdminListeners() field.ErrorList {
 				r.Spec.Configuration.AdminAPI,
```

Comment:
hmm externaldns and subdomain are two different things now? 🤔 or would it map to the same thing if e.g. external dns is used?

Reply:
They are different concepts at this point. The ExternalDNS is set by the provider, whereas the subdomain by the user. This PR keeps them as separate, except that they cannot both be provided (see webhook).

It's worth considering the interaction, alongside TLS, which requires a subdomain (for the external listener).